### PR TITLE
Remove niche prefs

### PR DESCRIPTION
**Pull request by LWinterberg**
Resolves: #5714

* Resolves: https://github.com/audacity/audacity/issues/4341
* Resolves: https://github.com/audacity/audacity/issues/4185 
* Resolves: https://github.com/audacity/audacity/issues/3511 
* Resolves: https://github.com/audacity/audacity/issues/4440 
* Resolves: https://github.com/audacity/audacity/issues/5666
* Resolves: https://github.com/audacity/audacity/issues/2880

----

* Removes "solo=none" option
* Removes timeline tooltips preference
* Removes dragging selection edges
* Removes some warnings
* Removes the preference to scroll to negative time
* Hides the scrub ruler option from the preferences.

The latter is intentional - Whether the scrub ruler is shown or not feels a bit more like enabling a toolbar than setting a preference. A full removal of scrub ruler and scrub toolbar doesn't appear either as Audacity crashes upon startup when I do.

### CODE REVIEW

File: libraries/lib-playable-track/PlayableTrack.cpp [hunk 1]
```diff
@@ -189,9 +189,9 @@ EnumSetting<SoloBehavior> TracksBehaviorsSolo{
...
       ByColumns,
-      { XO("Multi-track"), XO("Simple"),  XO("None") },
-      { wxT("Multi"),     wxT("Simple"), wxT("None") }
+      { XO("Multi-track"), XO("Simple")},
+      { wxT("Multi"),     wxT("Simple")}
    },
-   0, // "Simple"
-   { SoloBehaviorMulti, SoloBehaviorSimple, SoloBehaviorNone },
+   0, // "Multi-track"
+   { SoloBehaviorMulti, SoloBehaviorSimple },
```
Comment (Paul-Licameli) [inline libraries/lib-playable-track/PlayableTrack.cpp:196]: Test the case that you select "None" in a previous version of Audacity, then open this one, and confirm that the preference defaults back to Multi-track.

Step `EnumSettingBase::Migrate` in the debugger to see it happen.

File: src/AdornedRulerPanel.cpp [hunk 2]
```diff
@@ -740,9 +740,7 @@ class AdornedRulerPanel::CommonCell : public TrackPanelCell
    {
       // May come here when recording is in progress, so hit tests are turned
       // off.
-      TranslatableString tooltip;
-      if (mParent->mTimelineToolTip)
```
Comment (Paul-Licameli) [inline src/AdornedRulerPanel.cpp:744]: This commit leaves `mTimelineToolTip` unused, so amend it to remove the member variable.

File: src/AdornedRulerPanel.cpp [hunk 3]
```diff
@@ -2385,16 +2380,6 @@ void AdornedRulerPanel::OnSyncSelToQuickPlay(wxCommandEvent&)
    gPrefs->Flush();
 }
 
-#if 0
-void AdornedRulerPanel::OnTimelineToolTips(wxCommandEvent&)
-{
-   mTimelineToolTip = (mTimelineToolTip)? false : true;
-   gPrefs->Write(wxT("/QuickPlay/ToolTips"), mTimelineToolTip);
-   gPrefs->Flush();
-}
```
Comment (Paul-Licameli) [inline src/AdornedRulerPanel.cpp:2394]: Please delete too, the commented-out declaration of this function in the header file.

File: src/prefs/GUIPrefs.cpp [hunk 4]
```diff
@@ -192,15 +192,7 @@ void GUIPrefs::PopulateOrExchange(ShuttleGui & S)
 #endif
    }
    S.EndStatic();
-
-   S.StartStatic(XO("Timeline"));
-   {
-      S.TieCheckBox(XXO("Show Scrub Ruler"),
-                    {wxT("/QuickPlay/ScrubbingEnabled"),
-                     false});
```
Comment (Paul-Licameli) [inline src/prefs/GUIPrefs.cpp:203]: The preference can also still be enabled with a button in the scrubbing toolbar.

OK, you are the designer, but I think it's too difficult to discover the scrubbing feature.  There is no longer, either, any clue about it in the context menu of the time ruler.

Scrubbing still needs a satisfactory user interface.

About your commit comment:  I follow the convention that if it doesn't all fit on one line, I end the first line with ellipsis ... and then there is one blank line before the rest.

File: src/prefs/TracksBehaviorsPrefs.cpp [hunk 5]
```diff
@@ -112,10 +112,6 @@ void TracksBehaviorsPrefs::PopulateOrExchange(ShuttleGui & S)
       S.TieCheckBox(XXO("Use dialog for the &name of a new label"),
                     {wxT("/GUI/DialogForNameNewLabel"),
                      false});
-#ifdef EXPERIMENTAL_SCROLLING_LIMITS
```
Comment (Paul-Licameli) [inline src/prefs/TracksBehaviorsPrefs.cpp:118]: See `SCROLLING_LIMITS` in `src/Experimental.cmake`.  You delete the sole use of `EXPERIMENTAL_SCROLLING_LIMITS`, and that is where this preprocessor constant is defined.

See my comments there about why I implemented scrolling beyond zero:

```
   # Paul Licameli (PRL) 24 May 2015
   # Allow scrolling up to one half of a screenful beyond either end of the project,
   # if you turn on the appropriate Tracks preference.
   # This allows smooth-scrolling scrub to work more reasonably at the ends.
```

I'd forgotten that motivation, myself!

Does this give any reason to pause about deleting this feature?

I had certain ideas about scrubbing -- I don't remember all of them just now, but I think one of them was an alternative mode that would keep the play head in a fixed position but let you drag the wave under it.

Comment (Paul-Licameli) [reply src/prefs/TracksBehaviorsPrefs.cpp:118]: If not, then also delete this part of Experimental.cmake

File: src/ProjectWindow.cpp [hunk 6]
```diff
@@ -708,30 +706,6 @@ std::pair<int, int> ProjectWindow::ViewportSize() const
    return { width, height };
 }
 
-bool ProjectWindow::MayScrollBeyondZero() const
-{
-   auto pProject = FindProject();
-   if (!pProject)
-      return false;
-   auto &project = *pProject;
-   auto &scrubber = Scrubber::Get( project );
```
Comment (Paul-Licameli) [inline src/ProjectWindow.cpp:717]: This deletion also allows you to eliminate
`#include "Scrubbing.h"`
at the top of the file.  Please do that!

File: src/prefs/GUIPrefs.cpp (hunk 4 above)
Comment (LWinterberg) [reply src/prefs/GUIPrefs.cpp:203]: I did try to remove the entire scrubbing toolbar and scrub bar, however, doing so crashes Audacity upon startup as it reads the old preferences file. Any idea how to fix that?

Comment (Paul-Licameli) [reply src/prefs/GUIPrefs.cpp:203]: I could debug it if you wrote a draft PR, if you don't have the debugging experience.

But if you do all of that, how is anybody supposed to scrub at all?  How, besides using the Transport menu, awkwardly?

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @LWinterberg 's latest branch for this:

1) Tracks Behaviors prefs
image

**Comment by Paul-Licameli** [comment]
I raise the suggestion again:  Why not bring back a checkmark in the time ruler to turn on the scrub ruler, making that more discoverable?

**Comment by LWinterberg** [comment]
> Why not bring back a checkmark in the time ruler to turn on the scrub ruler, making that more discoverable?

that's a good idea

**Comment by Paul-Licameli** [comment]
I see this failed CI because the "Sanity" test crashes -- that just stars Audacity and immediately quits it.

**Comment by Paul-Licameli** [comment]
> I see this failed CI because the "Sanity" test crashes -- that just stars Audacity and immediately quits it.

I debugged that for you and pushed a commit onto your branch to fix it.

**Comment by Paul-Licameli** [comment]
I see is has been rebased, and there is just one new change -- to eliminate a lookup of the "Scrub" toolbar in Scrubbing.cpp.

Now I see, there is one other bit of code that can also be eliminated, in ProjectAudioManager.cpp.

I pushed one more commit to complete the work.

It can be done in another PR, and I ask again:  Do you want to put a context menu item in the time ruler to reenable the scrub ribbon?

**Review by Paul-Licameli — APPROVED** [review]

**Pull request merged**